### PR TITLE
Invisible Plantpots

### PR DESCRIPTION
**Pull request by Azrun**
[feature][bug-minor]

## About the PR 
Adds a bareplant plantpot to allow for plants to exist in the wild.  These plantpots use a reduced processing cycle (except for plants with a specialproc) to limit machine loop burden.  They automatically keep themselves watered.  They filter out a number of unwanted interactions (anchoring, tray checking, deconstruction).  Growth rate is reduced so it is subpar to a typical growing tray.

Will generate plant with defined plant datum path.  Can be provided a list of commutes to apply, also has flags for some commutes I thought would be more common (toxic, seedless, and terminator).

Adds component (and minor variations) to make arbitrary atoms able to grow plants.
- When attached to a movable atom the plant is added to the component owners vis_contents to allow for interaction.

NOTE:
Adds COMSIG_ATTACKBY signal generation to /mob/attackby() and /turf/simulated/floor/attackby.

## Why's this needed? 
- I want this for Artemis.
- Can be used to add plants to Azone and Prefabs.  Fun way to have _secret_ plants that need to be found and harvested...
- Resolves bug where seedless commute is not respected for seeded fruit.

### CODE REVIEW

**Comment by Azrun** [comment]
Corn!

**Comment by Gannets** [comment]
cool idea

**Comment by vocalpocal** [comment]
Pretty sure there are few sprites which don't work too well without pots, such as Creeper.

**Comment by Azrun** [comment]
> Pretty sure there are few sprites which don't work too well without pots, such as Creeper.

Adding simple sprite based on the plantpot soil to ease that. It otherwise reduces quality of other sprites.  So tool in the toolbox.

**Comment by Tarmunora** [comment]
note: attackby signal stuff will be in conflict with #5599

File: code/obj/machinery/plantpot.dm [hunk 1]
```diff
@@ -84,6 +84,136 @@
...
+	/// growth level to spawn with
+	var/spawn_growth = null
+	/// list of commuts to apply to plant
+	var/list/datum/plant_gene_strain/spawn_commuts = list()
+	/// apply seedless plant gene strain
+	var/seedless = null
+	/// apply terminator plant gene strain
+	var/terminator = null
+	/// apply toxic plant gene strain
+	var/toxic = null
```
Comment (Tarmunora) [inline code/obj/machinery/plantpot.dm:107]: why have the vars for those 3 commuts separate from the commut list?

**Comment by Azrun** [comment]
> note: attackby signal stuff will be in conflict with #5599

Perfect! Would love to delete that aspect of this PR!

File: code/obj/machinery/plantpot.dm (hunk 1 above)
Comment (Azrun) [reply code/obj/machinery/plantpot.dm:107]: Reduce barrier of entry to using the feature for those with limited hydroponics code knowledge. I have no qualms with it's removal.

**Review by Tarmunora — APPROVED** [review]

File: code/datums/components/arable_turf.dm [hunk 2]
```diff
@@ -0,0 +1,101 @@
+/datum/component/arable_turf
```
Comment (ZeWaka) [inline code/datums/components/arable_turf.dm:1]: at least a description of what this is would be nice

File: code/modules/hydroponics/plants_weed.dm [hunk 3]
```diff
@@ -203,11 +203,11 @@ ABSTRACT_TYPE(/datum/plant/weed)
 		if (POT.growth >= (P.harvtime + DNA.harvtime + 50) && prob(10) && !src.exploding)
 			src.exploding = 1
 			POT.visible_message("<span class='alert'><b>[POT]</b> begins to bubble and expand!</span>")
-			playsound(POT.loc, "sound/effects/bubbles.ogg", 50, 1)
+			playsound(POT, "sound/effects/bubbles.ogg", 50, 1)
```
Comment (ZeWaka) [inline code/modules/hydroponics/plants_weed.dm:206]: get turf?

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/plants_weed.dm (hunk 3 above)
Comment (Azrun) [reply code/modules/hydroponics/plants_weed.dm:206]: Commenting for completeness:
playsound calls get_turf() on first parameter

File: code/datums/components/arable_turf.dm (hunk 2 above)
Comment (Azrun) [reply code/datums/components/arable_turf.dm:1]: Then I'm going to add a description... then rename because it can be used on more than just turfs...

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**